lib/linters/localized_validation_message_linter.rb

```diff
@@ -0,0 +1,40 @@
+module RuboCop
+  module Cop
+    module IdentityIdp
+      class LocalizedValidationMessageLinter < RuboCop::Cop::Cop
+        MSG = 'Use Proc.new when translating validation message'.freeze
+
+        RESTRICT_ON_SEND = [
+          :validate,
+          :validates,
+          :validates!,
+          :validates_with,
+          :validates_absence_of,
+          :validates_acceptance_of,
+          :validates_confirmation_of,
+          :validates_exclusion_of,
+          :validates_format_of,
+          :validates_inclusion_of,
+          :validates_length_of,
+          :validates_numericality_of,
+          :validates_presence_of,
+          :validates_size_of,
+        ].freeze
+
+        def_node_matcher :translated_validation_message?, <<~PATTERN
+          (send nil? /^validate|validates|validates!$/ _ (hash (pair _ (hash (pair (sym :message) (send (const nil? :I18n) :t ...))))))
```

Comment:
s-expressions oh boy

Reply:
> s-expressions oh boy

There were very much moments of [confusion](https://i.kym-cdn.com/photos/images/original/000/234/765/b7e.jpg) in coming up with these 😅